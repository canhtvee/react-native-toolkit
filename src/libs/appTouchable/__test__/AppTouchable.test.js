import React from 'react';
import {Text} from 'react-native';

import {render, fireEvent, cleanup} from '@testing-library/react-native';

import {AppTouchable} from '../index';

/**
 * Run test command
 * yarn test -u -t="AppTouchable"
 */

afterAll(cleanup);

describe('AppTouchable', () => {
  test('should render correctly', () => {
    const tree = render(
      <AppTouchable activeOpacity={0.8} testID="pressable">
        <Text>TestingAppTouchable</Text>
      </AppTouchable>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('should invoke onPressMock if pressed', () => {
    const onPressMock = jest.fn();
    const {getByTestId} = render(
      <AppTouchable onPress={onPressMock} testID="pressable">
        <Text>Press AppTouchable</Text>
      </AppTouchable>,
    );
    //
    const pressable = getByTestId('pressable');
    fireEvent.press(pressable);
    expect(onPressMock).toHaveBeenCalled();
  });
});
