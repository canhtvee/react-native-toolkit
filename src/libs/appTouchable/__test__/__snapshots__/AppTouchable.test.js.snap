// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTouchable should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {
        "opacity": 1,
      },
    ]
  }
  testID="pressable"
>
  <Text>
    TestingAppTouchable
  </Text>
</View>
`;
